test/Microsoft.HttpRepl.Tests/HttpStateTests.cs

```diff
@@ -169,11 +169,80 @@ public void GetEffectivePath_ProperConcatenation(string baseUriString, string pa
         {
             Uri baseUri = new Uri(baseUriString);
 
-            Uri result = HttpState.GetEffectivePath(baseUri, pathSections, specifiedPath);
+            Uri result = HttpState.GetEffectivePath(baseUri, pathSections, specifiedPath, null);
 
             Assert.Equal(expectedResult, result.ToString(), StringComparer.OrdinalIgnoreCase);
         }
 
+        [Fact]
+        public void GetEffectivePath_AppendQueryString_SingleCase()
+        {
+            string baseUriString = "https://github.com/";
+            string pathSections = "dir1";
+            string specifiedPath = "dir2";
+            var queryString = new Dictionary<string, IEnumerable<string>>()
+            {
+                ["key"] = new List<string>() { "value"}
+            }; 
+            Uri baseUri = new Uri(baseUriString);
+
+            Uri result = HttpState.GetEffectivePath(baseUri, pathSections, specifiedPath, queryString);
+
+            Assert.Equal("https://github.com/dir1/dir2?key=value", result.ToString(), StringComparer.OrdinalIgnoreCase);
```

Comment:
Can you lift the string literal here to a local variable at the top (probably something like `expectedPath`) so they're near the construction of the parameters for easy comparison? Same for the other test methods below.

Reply:
Done